build.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+N_CORES=$(sysctl -n hw.ncpu) # macOS
+# TODO: get cores linux
+
+# ESP-IDF v3.x
+esp_idf_v3_build_native() {
+    cp arduinojason.mk components/ArduinoJson/component.mk
+    make all -j$N_CORES;
+    rm components/ArduinoJson/component.mk
+}
+
+# build docker v3.x
+esp_idf_v3_build_docker() {
+    N_CORES_DOCKER=$(docker run --rm espressif/idf:v3.3.4 /bin/bash -c "grep processor /proc/cpuinfo | wc -l; exit" | tail -1)
+    cp arduinojason.mk components/ArduinoJson/component.mk
+    docker run --rm -v $PWD:/project -w /project espressif/idf:v3.3.4 make all -j$N_CORES_DOCKER;
```

Comment:
I've heard that `N_CORES_DOCKER+1` can be faster because one task can be blocked on I/O, but I don't know whether it's true

Reply:
`make all` compiles a lot of stuff and is not necessary every build. I usually just run `make`, and `make app` in the server (doesn't build the bootloader etc)

Reply:
Using a volume instead of a dockerfile with ADD slows compilation wayyyy down for me on Mac.

Reply:
Tried this with and without `all` as native, doesn't change anything, which is great

Reply:
There's a bug with caching. I'm looking into it.